crates/signer/src/signer.rs

```diff
@@ -3,17 +3,19 @@ use alloy::primitives::Address;
 use alloy::signers::aws::{AwsSigner, AwsSignerError};
 use alloy::signers::local::PrivateKeySigner;
 use eth_keystore::decrypt_key;
+use serde::{Deserialize, Serialize};
 use std::path::Path;
 use thiserror::Error;
 use url::Url;
 
 /// Represents the input params to create a signer
-#[derive(Debug)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
 pub enum Config {
     /// Hexadecimal private key
-    PrivateKey(String),
+    PrivateKey { private_key: String },
     /// Keystore path and password
-    Keystore(String, String),
+    Keystore { path: String, password: String },
```

Comment:
I'm not sure we can do this. Wouldn't this be a breaking change? Can users unpack this enum even if tuple items aren't marked as `pub`?